cellprofiler/gui/pipelinecontroller.py

```diff
@@ -3136,6 +3137,40 @@ def remote_debug(evtlist=None):
 
         wx.Yield()  # This allows cancel events to remove other exceptions from the queue.
 
+    def sentry_pack_pipeline(self, err_module_name="", err_module_num=0):
+        from sentry_sdk import utils, serializer, push_scope
+        utils.MAX_STRING_LENGTH = 15000
+        serializer.MAX_DATABAG_BREADTH = 25
+        # push_scope will attach data to the next event only
+        with push_scope() as scope:
+            with io.StringIO() as fp:
+                self.__pipeline.dump(fp, save_image_plane_details=False, sanitize=False)
+                pipeline_parts = fp.getvalue().split("\n\n")
+                pipeline_info = {
+                    "error_module": f"{err_module_num:02d}_{err_module_name}",
+                    "header": pipeline_parts.pop(0),
+                    "modules": [module.module_name for module in self.__pipeline.modules(exclude_disabled=False)]
+                }
+                scope.set_context("Pipeline_Info", pipeline_info)
+                index = 1
+                buffer = 0
+                buffer_dict = {}
+                # Sentry hard limits context content to >~10000 characters per container.
+                # To avoid this we send in chunks if the pipeline is large.
+                for idx, part in enumerate(pipeline_parts):
+                    idx += 1
+                    keyname = f"Pipeline_{index:02d}"
```

Comment:
Is it possible to have multiple pipelines? 🤔

Reply:
For future reference to anyone reading - Sentry has a cap on how much data can be sent in a single context object, so we split long pipelines up into chunks.